File: apps/lib/models/conversion_status.dart
class ConversionStatus {
  final int conversionId;
  final String status; // pending, processing, completed, failed, cancelled
  final double progress; // 0.0 - 1.0
  final int processedBytes;
  final int totalBytes;
  final String? error;
  final String outputPath;
  final String? mode;
  final String? videoEncoder;
  final String? probeWarning;

  ConversionStatus({
    required this.conversionId,
    required this.status,
    required this.progress,
    this.processedBytes = 0,
    this.totalBytes = 0,
    this.error,
    required this.outputPath,
    this.mode,
    this.videoEncoder,
    this.probeWarning,
  });

  factory ConversionStatus.fromJson(Map<String, dynamic> json) {
    return ConversionStatus(
      conversionId: json['conversion_id'] ?? 0,
      status: json['status'] ?? 'pending',
      progress: (json['progress'] ?? 0.0).toDouble(),
      processedBytes: json['processed_bytes'] ?? 0,
      totalBytes: json['total_bytes'] ?? 0,
      error: json['error'],
      outputPath: json['output_path'] ?? '',
      mode: json['mode'] as String?,
      videoEncoder: json['video_encoder'] as String?,
      probeWarning: json['probe_warning'] as String?,
    );
  }
}

File: native/models/conversion_status.go
package models

type ConversionStatus struct {
	ConversionID   uintptr `json:"conversion_id"`
	Status         string  `json:"status"`   // pending, processing, completed, failed, cancelled
	Progress       float64 `json:"progress"` // 0.0 - 1.0
	ProcessedBytes int64   `json:"processed_bytes"`
	TotalBytes     int64   `json:"total_bytes"`
	Error          string  `json:"error,omitempty"`
	OutputPath     string  `json:"output_path"`
	Mode           string  `json:"mode,omitempty"`          // remux, cpu_encode, gpu_encode, audio_encode, gif_encode
	VideoEncoder   string  `json:"video_encoder,omitempty"` // copy, libx264, h264_nvenc, etc.
	ProbeWarning   string  `json:"probe_warning,omitempty"`
}

type ConversionExecutionInfo struct {
	Mode         string
	VideoEncoder string
	ProbeWarning string
	OutputPath   string
}
